### PR TITLE
eventpb: make the Timestamp field an int64

### PR DESCRIPTION
**Pull request by knz**
First commit from  #58066.

While exploring how to use the data in practice, I noticed that the
timestamp was encoded as a string in the JSON output, and that was
making the events hard to parse and to use. There are many
more functions that can operate on a number.

Moreover, the integer representation is just more compact.

Release note (api change):  The 'Timestamp' field of structured
notable events is now numeric, and encodes a number of nanoseconds
since the unix epoch. (Note that this API has not yet been published
in a released version of CockroachDB. The release note exists only to
track the list of relevant items for the doc project.)

### CODE REVIEW

**Review by itsbilal — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r1, 8 of 8 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/58070#-:-MP4zm61D9EP_-WLzMmm:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained

**Comment by knz** [comment]
TFYR!

bors r=itsbilal
